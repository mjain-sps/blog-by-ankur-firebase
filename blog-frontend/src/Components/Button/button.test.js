import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ButtonComponent from "./button.component";

describe("check the button component snapshot", () => {
  const dummyProps = {
    type: "button",
    onClick: jest.fn(),
    theme: "primary",
    width: "50%",
  };
  const dummyProps2 = {
    type: "button",
    onClick: jest.fn(),
    theme: "primary",
  };
  test("check snapshot if width present", () => {
    const { container } = render(<ButtonComponent {...dummyProps} />);
    expect(container).toMatchSnapshot();
  });
  test("check snapshot if width is null", () => {
    const { container } = render(<ButtonComponent {...dummyProps2} />);
    expect(container).toMatchSnapshot();
  });
});
