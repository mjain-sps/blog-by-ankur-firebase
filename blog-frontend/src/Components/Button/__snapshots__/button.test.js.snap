// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the button component snapshot check snapshot if width is null 1`] = `
<div>
  <button
    class="sc-bdnxRM iVoCAH"
    type="button"
    width="100"
  />
</div>
`;

exports[`check the button component snapshot check snapshot if width present 1`] = `
<div>
  <button
    class="sc-bdnxRM hwEZbu"
    type="button"
    width="50%"
  />
</div>
`;
